### PR TITLE
Updating repository references to new repository name

### PR DESCRIPTION
**Pull request by jjbailey-io**
Renaming repository to rules-engine.

### CODE REVIEW

File: README.md
Comment (ShaneDuncan602) [inline README.md:1]: Rules Protocol is still the title in this file. Are we leaving that?
We also have a bunch of references to Rule Protocol Client in here.

File: docs/OVERVIEW.md
Comment (ShaneDuncan602) [inline docs/OVERVIEW.md:1]: This also says Rules Protocol

File: docs/userGuides/GLOSSARY.md
Comment (ShaneDuncan602) [inline docs/userGuides/GLOSSARY.md:1]: Contains Rules Protocol

**Review by ShaneDuncan602 — APPROVED** [review]
Looks good.

**Review by excalq — APPROVED** [review]

**Review by bcamp136 — APPROVED** [review]

**Pull request merged**